packages/utils/shared/deep-map.ts

```diff
@@ -1,11 +1,11 @@
-import { isObjectLike } from 'lodash-es';
+import { isObjectLike, isPlainObject } from 'lodash-es';
 
 export function deepMap(object: any, iterator: (value: any, key: string | number) => any, context?: any): any {
 	if (Array.isArray(object)) {
 		return object.map(function (val, key) {
 			return isObjectLike(val) ? deepMap(val, iterator, context) : iterator.call(context, val, key);
```

Comment:
Should this `isObjectLike` be replaced by `isPlainObject` too? shouldnt make sense for it to attempt to loop over `new Date()` if that value is in an array 🤔

Reply:
Technically yes but I did rewrite deepMap anyways in #25216 so that would improve it.